src/classes/BGE_BatchGiftEntry_UTIL.cls

```diff
@@ -116,34 +119,47 @@ public with sharing class BGE_BatchGiftEntry_UTIL {
 
                         BDI_FieldMapping fieldMapping =
                                 fieldMappingByDevName.get(fieldMappingName);
-                        if (fieldMapping == null) {
+
+                        Boolean isObjectMappedLookup = mappingService
+                                .importedRecordFieldNames.contains(
+                                fieldMappingName.toLowerCase());
+
+                        if (fieldMapping == null && !isObjectMappedLookup) {
                             continue;
                         }
-
+                        String sourceFieldName = fieldMapping != null ?
+                                fieldMapping.Source_Field_API_Name :
+                                isObjectMappedLookup ?
+                                        fieldMappingName : null;
                         // Add the source fields used in the Template
-                        if (!fieldApiNames.contains(
-                                fieldMapping.Source_Field_API_Name.toLowerCase())) {
-                            fieldApiNames.add(
-                                    fieldMapping.Source_Field_API_Name.toLowerCase());
+                        if (!fieldApiNames.contains(sourceFieldName.toLowerCase())) {
+                            fieldApiNames.add(sourceFieldName.toLowerCase());
```

Comment:
I don't think we should make this change RIGHT now, but for the future - a combination of sourceFieldName + targetFieldName might be a better key here, to account for possible cases where the same DI field is mapped to multiple locations (as is frequently the case with the Record Imported fields)